tests/optimizer/test-function-eliminator-double-parsed-correctly-output.js

```diff
@@ -0,0 +1,14 @@
+// EMSCRIPTEN_START_ASM
+var asm = (function(global, env, buffer) {
+ "use asm";
+ var e = 0;
+ 
+// EMSCRIPTEN_START_FUNCS
+function a() {
+ var c = 0.0;
```

Comment:
this implies to me that running the DFE tool will finalize `+0` into `0.0`. Is that right?

If so, that's a problem, as running any other JS optimizer pass afterwards will collapse `0.0` into `0`, losing the fact that it is a double.

Instead, the DFE tool should leave `+0` as `+0`, same as the js optimizer does when it isn't given "last" as an indication that no further passes will be run.